server/controllers/stock/index.js

```diff
@@ -298,96 +298,91 @@ async function createInventoryAdjustment(req, res, next) {
-        depot_uuid : db.bid(movement.depot_uuid),
-        document_uuid : db.bid(negativeAdjustmentUuid),
-        quantity : lot.oldQuantity,
-        unit_cost : lot.unit_cost,
-        date : new Date(movement.date),
-        entity_uuid : movement.entity_uuid,
-        is_exit : 1,
-        flux_id : core.flux.INVENTORY_RESET,
-        description : movement.description,
-        user_id : req.session.user.id,
-        period_id : periodId,
-      };
-      trx.addQuery('INSERT INTO stock_movement SET ?', reverseMovementObject);
-    });
-
-    negativeQuantities.forEach(lot => {
-      const reverseMovementObject = {
-        uuid : db.bid(uuid()),
-        lot_uuid : db.bid(lot.uuid),
-        depot_uuid : db.bid(movement.depot_uuid),
-        document_uuid : db.bid(positiveAdjustmentUuid),
-        quantity : lot.oldQuantity * -1,
-        unit_cost : lot.unit_cost,
-        date : new Date(movement.date),
-        entity_uuid : movement.entity_uuid,
-        is_exit : 0,
-        flux_id : core.flux.INVENTORY_RESET,
-        description : movement.description,
-        user_id : req.session.user.id,
-        period_id : periodId,
-      };
-      trx.addQuery('INSERT INTO stock_movement SET ?', reverseMovementObject);
+      if (difference < 0) {
+        // we have to do a stock exit movement
+        // we must substract the |difference| to the current quantity
+        countNeedDecrease++;
+        const movementObject = {
+          uuid : db.bid(uuid()),
+          lot_uuid : db.bid(lot.uuid),
+          depot_uuid : db.bid(movement.depot_uuid),
+          document_uuid : db.bid(uniqueAdjustmentUuid),
+          quantity : Math.abs(difference),
+          unit_cost : lot.unit_cost,
+          date : new Date(movement.date),
+          entity_uuid : movement.entity_uuid,
+          is_exit : 1,
+          flux_id : core.flux.INVENTORY_ADJUSTMENT,
+          description : movement.description,
+          user_id : req.session.user.id,
+          period_id : periodId,
+        };
+        adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 1 }));
+        trx.addQuery('INSERT INTO stock_movement SET ?', movementObject);
+      } else {
+        // we must do increase the current quantity by the |difference|
+        countNeedIncrease++;
+        const movementObject = {
+          uuid : db.bid(uuid()),
+          lot_uuid : db.bid(lot.uuid),
+          depot_uuid : db.bid(movement.depot_uuid),
+          document_uuid : db.bid(uniqueAdjustmentUuid),
+          quantity : Math.abs(difference),
+          unit_cost : lot.unit_cost,
+          date : new Date(movement.date),
+          entity_uuid : movement.entity_uuid,
+          is_exit : 0,
+          flux_id : core.flux.INVENTORY_ADJUSTMENT,
+          description : movement.description,
+          user_id : req.session.user.id,
+          period_id : periodId,
+        };
+        adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 0 }));
```

Comment:
Just FYI, you can do the `_.assign()` operation with vanilla javascript.  Like this:
```js
var x = _.assign({}, lot, {difference, is_exit: 0 });
// is equivalent to 
var y = { ...lot, difference, is_exit };
```

Reply:
I think this may end up being a lot of information as written.  Would it be possible to store only the `movement.uuid` and the `difference`?  Then, when you are rendering the receipt, you simply stitch them together with while displaying the receipt?

Reply:
Actually, if you did that, you wouldn't need a JSON column.  Just two columns - `row_uuid` and `difference`.  You could then do the `JOIN` in MySQL!